caiman_napari/algorithms/cnmfe.py

```diff
@@ -0,0 +1,214 @@
+                    "traceback": None
+                }
+            )
+            print(d)
+
+    except:
+        d = {"success": False, "traceback": traceback.format_exc()}
+    # Add dictionary to output column of series
+    df.loc[df['uuid'] == uuid, 'outputs'] = [d]
+    # save dataframe to disc
+    df.to_pickle(batch_path)
+
+def load_output(viewer, batch_item: pd.Series):
+    print('loading outputs of CNMFE')
+    path = batch_item["outputs"].item()["cnmfe_outputs"]
+    params = batch_item["params"].item()
+    uuid = batch_item['uuid']
+
+    if not params['do_cnmfe']:
+        cn_filter = pd.read_pickle(path[0])
+        viewer.add_image(cn_filter)
+        pnr_filter = pd.read_pickle(path[1])
+        viewer.add_image(pnr_filter)
+    else:
+        cnmfe_obj = load_CNMF(path)
+        print(cnmfe_obj)
+        dims = cnmfe_obj.dims
+        if dims is None:
+            dims = cnmfe_obj.estimates.dims
+
+        dims = dims[1], dims[0]
+
+        contours_good = caiman_get_contours(
+            cnmfe_obj.estimates.A[:, cnmfe_obj.estimates.idx_components],
+            dims,
+            swap_dim=True
+        )
+
+        colors_contours_good_edge = auto_colormap(
+            n_colors=len(contours_good),
+            cmap='hsv',
+            output='mpl',
+        )
+        colors_contours_good_face = auto_colormap(
+            n_colors=len(contours_good),
+            cmap='hsv',
+            output='mpl',
+            alpha=0.0,
+        )
+
+        contours_good_coordinates = [_organize_coordinates(c) for c in contours_good]
+        viewer.add_shapes(
+            data=contours_good_coordinates,
+            shape_type='polygon',
+            edge_width=0.5,
+            edge_color=colors_contours_good_edge,
+            face_color=colors_contours_good_face,
+            opacity=0.7,
+        )
+        if cnmfe_obj.estimates.idx_components_bad is not None and len(cnmfe_obj.estimates.idx_components_bad) > 0:
+            contours_bad = caiman_get_contours(
+                cnmfe_obj.estimates.A[:, cnmfe_obj.estimates.idx_components_bad],
+                dims,
+                swap_dim=True
+            )
+
+            contours_bad_coordinates = [_organize_coordinates(c) for c in contours_bad]
+
+            colors_contours_bad_edge = auto_colormap(
+                n_colors=len(contours_bad),
+                cmap='hsv',
+                output='mpl',
+            )
+            colors_contours_bad_face = auto_colormap(
+                n_colors=len(contours_bad),
+                cmap='hsv',
+                output='mpl',
+                alpha=0.0
+            )
+
+            viewer.add_shapes(
+                data=contours_bad_coordinates,
+                shape_type='polygon',
+                edge_width=0.5,
+                edge_color=colors_contours_bad_edge,
+                face_color=colors_contours_bad_face,
+                opacity=0.7,
+            )
+
+def _organize_coordinates(contour: dict):
```

Comment:
Maybe put this in `caiman_napari.utils` instead of in both the `cnmf` and `cnmfe` modules.

Reply:
In pycharm you can automatically move a function to another module. I think you can just click on function, then press F6 to move it.